client/me/form-base/index.js

```diff
@@ -39,16 +40,21 @@ export default {
 		return {
 			redirect: false,
 			submittingForm: false,
+			formsSubmitting: {},
 			changingUsername: false,
 			usernameAction: 'new',
 			showNotice: false,
 		};
 	},
 
-	showNotice: function () {
+	showNotice: function ( formName ) {
 		if ( this.props.userSettings.initialized && this.state.showNotice ) {
 			notices.clearNotices( 'notices' );
-			notices.success( this.props.translate( 'Settings saved successfully!' ) );
+			notices.success(
+				this.props.translate( '%s Settings saved successfully!', {
```

Comment:
Let's add a comment here for translators so they have extra info about what can be displayed here: https://github.com/Automattic/wp-calypso/tree/trunk/packages/i18n-calypso#options

Alternatively, not sure if this can cause any issue in other languages. Some translators may have have some difficulties translating a dynamic form name. For instance, `Account settings` is translated as `Ajustes de cuenta` in Spanish, but just `Account` should be translated as `Cuenta` and `% Settings` can be translated as just `Ajustes %` resulting in a wrong `Ajustes Cuenta` translation.

It may be better to explicitly define the 2 possible translations here, even if that means the code is a bit more repetitive.

```js
if ( formName === 'Account' ) {
   this.props.translate( 'Account settings saved successfully!' );
} else if if ( formName === 'Interface' ) {
   this.props.translate( 'Interface settings saved successfully!' );
}
```

cc @Automattic/i18n

Reply:
The only thing is I don't want to couple the FormBase to particular form names. I made the translation because when you save x2 forms at once, it wasn't clear in the UI which form was saved if we left the original "generic" text.

Is there a way around which doesn't couple unrelated form names to the Base Form Mixin?

Reply:
> if you have unsaved changes in both forms, then save one of them and reload the page, there is no alert warning you that the second form has unsaved changes.

I didn't realise that was a regression. It looks to me like the FormBase mixin clears all unsaved changes when the component unmounts.

https://github.com/Automattic/wp-calypso/blob/0f703bed9ed70c1a09da2da74dbe6ab3cf6a19b0/client/me/form-base/index.js#L21-L22

Reply:
I see that you've already reverted back to the generic message, but I wanted to confirm that @mmtr's example with the full strings is usually the preferred approach when the placeholder is supposed to be replaced with a noun. Otherwise, it could make it impossible to translate for some languages with grammatical gender system, where the noun is supposed to alter other parts of the sentence.